lisa/sut_orchestrator/azure/features.py

```diff
@@ -668,6 +674,75 @@ def _initialize(self, *args: Any, **kwargs: Any) -> None:
             len(all_nics) - self.origin_extra_synthetic_nics_count - 1
         )
 
+    def create_route_table(
+        self,
+        nic_name: str,
+        route_name: str,
+        subnet_mask: str,
+        dest_hop: str,
+        em_first_hop: str = "",
+        next_hop_type: str = "",
+    ) -> None:
+        # some quick checks that the subnet mask looks correct
+        check_mask = self.__ipv4_mask_check_regex.match(subnet_mask)
+        assert_that(check_mask).described_as(
+            "subnet mask should be prefix format X.X.X.X/YY"
+        ).is_not_none()
+
+        azure_platform: AzurePlatform = self._platform  # type: ignore
+        # Step 1: create the route table, apply comes later.
+        route_table = self._do_create_route_table(
+            em_first_hop,
+            subnet_mask=subnet_mask,
+            nic_name=nic_name,
+            route_name=route_name,
+            next_hop_type=next_hop_type,
+            dest_hop=dest_hop,
+        )
+        # Step 2: Get the virtual networks in the resource group.
+        vnets: Dict[str, List[str]] = get_virtual_networks(
+            azure_platform, self._resource_group_name
+        )
+        # get the subnets in the virtual network
+        vnet_id = ""
+        subnet_ids = []
+        assert_that(vnets.items()).described_as(
+            "There is more than one virtual network in this RG!"
+            " This RG is setup is unexpected, test cannot infer which VNET to use."
+            "Check if LISA has changed it's test setup logic, verify if the "
+            "DPDK test suite needs to be modified."
+        ).is_length(1)
+        # get the subnets for the virtual network in the test RG.
+        for vnet, subnets in vnets.items():
```

Comment:
it's already asserted only 1 item, so use vnets[0] is clearer than a loop..

Reply:
This pattern was cribbed from elsewhere in the file 😜 
I was trying to follow what was already there, fixing now though...

Reply:
I get why they did it the other way. mypy wants a type hint when you unroll the tuple after casting the dict_items into a list. The old way you don't get a type complaint for some reason.